pkg/codegen/dotnet/gen.go

```diff
@@ -465,15 +474,41 @@ func (pt *plainType) genInputProperty(w io.Writer, prop *schema.Property, indent
 		}
 		printObsoleteAttribute(w, prop.DeprecationMessage, indent)
 
-		// Note that we use the backing field type--which is just the property type without any nullable annotation--to
-		// ensure that the user does not see warnings when initializing these properties using object or collection
-		// initializers.
-		fmt.Fprintf(w, "%spublic %s %s\n", indent, backingFieldType, propertyName)
-		fmt.Fprintf(w, "%s{\n", indent)
-		fmt.Fprintf(w, "%s    get => %[2]s ?? (%[2]s = new %[3]s());\n", indent, backingFieldName, backingFieldType)
-		fmt.Fprintf(w, "%s    set => %s = value;\n", indent, backingFieldName)
+		switch prop.Type.(type) {
+		case *schema.ArrayType, *schema.MapType:
+			// Note that we use the backing field type--which is just the property type without any nullable annotation--to
+			// ensure that the user does not see warnings when initializing these properties using object or collection
+			// initializers.
+			fmt.Fprintf(w, "%spublic %s %s\n", indent, backingFieldType, propertyName)
+			fmt.Fprintf(w, "%s{\n", indent)
+			fmt.Fprintf(w, "%s    get => %[2]s ?? (%[2]s = new %[3]s());\n", indent, backingFieldName, backingFieldType)
+		default:
+			fmt.Fprintf(w, "%spublic %s? %s\n", indent, backingFieldType, propertyName)
+			fmt.Fprintf(w, "%s{\n", indent)
+			fmt.Fprintf(w, "%s    get => %s;\n", indent, backingFieldName)
+		}
+		if prop.Secret {
+			fmt.Fprintf(w, "%s    set\n", indent)
+			fmt.Fprintf(w, "%s    {\n", indent)
+			// Since we can't directly assign the Output from CreateSecret to the property, use an Output.All or
+			// Output.Tuple to enable the secret flag on the data.
```

Comment:
Maybe its just me that doesn't know enough about the dotnet sdk semantics here but it was a headscratcher to figure out how this actually resulted in the value being a secret.
```suggestion
			// Output.Tuple to enable the secret flag on the data, since if any inputs passed to either
			// were secrets, the output will be a secret.
```